plan/join_reorder.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"github.com/ngaut/log"
+	"github.com/pingcap/tidb/ast"
+	"github.com/pingcap/tidb/expression"
+)
+
+func tryToGetJoinGroups(j *Join) ([]LogicalPlan, bool) {
```

Comment:
It returns a slice of Plan in a single group, so the name should be 'tryToGetJoinGroup'.